### DISCUSSION TITLE
Prometheus /metric endpoint

### DISCUSSION THREAD

**Discussion by nickvazz** [dask/dask#7443] [Q&A] [answered]
Hello! 

I am trying to access the [Prometheus /metric endpoint](https://distributed.dask.org/en/latest/http_services.html?highlight=Prometheus#worker-http) but there seems to be nothing there. I am using Dask & Distributed 2.20 (can potentially upgrade - running in production) for the scheduler via the [docker images](https://github.com/dask/dask-docker). Is there some library that is required to be installed for these endpoints to be generated? 

Thank you!

**Comment by jrbourbeau** [comment, marked as answer]
I believe that `prometheus_client` needs to be installed

**Reply by nickvazz** [reply]
Just for clarification, if I add that as an `EXTRA_PIP_PACKAGES` ([noted here](https://github.com/dask/dask-docker#environment-variables)), the endpoint should then exist?

**Reply by jrbourbeau** [reply]
Yes, that should be the case. Locally, I if uninstall `prometheus_client` then when I go to the `/metrics` (note that it's `metrics` not `metric`) then I get error messages like

```python
`ModuleNotFoundError: No module named 'prometheus_client'`
```

After installed `prometheus_client` I get content when hitting the `/metrics` endpoint. 

As you mentioned, adding `prometheus_client` to `EXTRA_PIP_PACKAGES` will make sure `prometheus_client` is installed inside the docker image you're using

**Reply by jrbourbeau** [reply]
Though it'd be great if you could confirm this is also the case on your system

**Reply by nickvazz** [reply]
Will do! Will be trying this afternoon and will let you know.

**Reply by nickvazz** [reply]
Worked as suggested! Thank you for the quick replies

**Reply by nickvazz** [reply]
Opened a PR to add the package requirement to the docs https://github.com/dask/dask/pull/7454
